### PR TITLE
show all tags

### PR DESCRIPTION
**Pull request by johnmanjiro13**
fix: #30 

I added --show-all-tags flag to autoscaling and instance commands.
ec2-search always give preference --show-all-tags between --show-all-tags and -T.

itertools was introduced for using `unique`.

### CODE REVIEW

**Comment by johnmanjiro13** [comment]
Some unrelated files are changed (ex. utils.rs) because I fixed some clippy's errors.

File: src/autoscaling.rs [hunk 1]
```diff
@@ -52,10 +55,22 @@ pub async fn matcher(opt: AutoScalingGroupOpt) {
...
+    let tag_column: Vec<String> = if opt.show_all_tags {
+        asg.iter()
+            .map(|t| t.tags.iter().map(|ot| ot.key.to_string()).collect())
+            .collect::<Vec<Vec<String>>>()
+            .into_iter()
+            .flatten()
+            .collect::<Vec<String>>()
+            .into_iter()
+            .unique()
+            .collect()
```
Comment (mocyuto) [inline src/autoscaling.rs:68]: Do not need use `collect` in method chains, use collect at last like below.
```rust
.map(|t| t.tags.iter().map(|ot| ot.key.to_string()))
            .flatten()
            .unique()
            .collect()
```

File: src/instance.rs [hunk 2]
```diff
@@ -58,10 +61,22 @@ pub async fn matcher(opt: InstanceOpt) {
...
+        instances
+            .iter()
+            .map(|t| t.tags.iter().map(|ot| ot.key.to_string()).collect())
+            .collect::<Vec<Vec<String>>>()
+            .into_iter()
+            .flatten()
+            .collect::<Vec<String>>()
+            .into_iter()
+            .unique()
+            .collect()
```
Comment (mocyuto) [inline src/instance.rs:74]: same as above

File: src/autoscaling.rs (hunk 1 above)
Comment (johnmanjiro13) [reply src/autoscaling.rs:64]: Oops, I didn't know that.
Thanks!

Comment (johnmanjiro13) [reply src/autoscaling.rs:64]: I fixed it e004b19

**Review by mocyuto — APPROVED** [review]
Thanks! LGTM

**Pull request merged**